tfx_bsl/beam/run_inference.py

```diff
@@ -697,81 +785,111 @@ def setup(self):
           signature_def.method_name)
     super(_BatchPredictDoFn, self).setup()
 
-  def _check_elements(
-      self, elements: List[Union[tf.train.Example,
-                                 tf.train.SequenceExample]]) -> None:
+  def _check_elements(self) -> None:
     pass
 
   def _post_process(
-      self, elements: Union[Sequence[tf.train.Example],
-                            Sequence[tf.train.SequenceExample]],
+      self, elements: Mapping[Any, Any],
       outputs: Mapping[Text, np.ndarray]
   ) -> Iterable[prediction_log_pb2.PredictLog]:
+    if not self._io_tensor_spec.input_tensor_types:
+      raise ValueError('No valid tensor types.')
+    input_tensor_names = self._io_tensor_spec.input_tensor_names
     input_tensor_alias = self._io_tensor_spec.input_tensor_alias
+    input_tensor_types = self._io_tensor_spec.input_tensor_types
     signature_name = self._signatures[0].name
-    batch_size = len(elements)
+
+    if len(input_tensor_alias) != len(input_tensor_names):
+      raise ValueError('Expected to have one name and one alias per tensor')
+
+    include_request = True
+    if len(input_tensor_names) == 1:
+      serialized_examples, = elements.values()
```

Comment:
Shall we also check the type of elements.values is string/bytes?

Reply:
It's checked in extract form record batch